### ISSUE TITLE
AndroidNotificationChannel.LockScreenVisibility.Private not working

### ISSUE THREAD

**Issue by Maicolo** [Unity-Technologies/com.unity.mobile.notifications#35]
Hello guys, while testing the lock screen visibility levels on Android I realised the 'Private' level doesn't seem to work.

Looking into the source it seems that 'AndroidNotificationChannel.LockScreenVisibility' enum sets the 'Private' label equal to '-1000'.

However according to [Android documentation](https://developer.android.com/reference/androidx/core/app/NotificationCompat#VISIBILITY_PRIVATE)   the value for the 'private' state of the lock screen visibility is 0 (0x00000000).

**Comment by vincent-unity** [comment]
Thanks for reporting this. 
I don't know why it's -1000 at the beginning, but you're right according to the android doc. Will have a check.

**Comment by vincent-unity** [comment]
I looked into this issue today, would like to share what I found.

1. `-1000` is the default value of the `LockscreenVisibility` on a channel, which stands for `public`. If you create a channel without setting this, you can see `NotificationChannel.getLockscreenVisibility()` will return -1000. If you check in the `System settings`, you can see the channel is set to `show all notifications ...` by default.

2. NotificationChannel.setLockscreenVisibility() won't work from code anyway. check the comment [here](https://developer.android.com/reference/android/app/NotificationChannel#setLockscreenVisibility(int)) by searching`Only modifiable by the system and notification ranker.`. This means the value can only be changed from `System settings` or by `Android Notification Ranking Service`. I did a quick experiment by doing below
```
    protected void createNotificationChannel(NotificationManager notificationManager)
    {
        NotificationChannel channel = new NotificationChannel(k_ChannelId, "Test Default", NotificationManager.IMPORTANCE_DEFAULT);

        Log.w("Create Channel", String.format("Default lockscreen: %d", channel.getLockscreenVisibility()));
        channel.setLockscreenVisibility(Notification.VISIBILITY_PRIVATE);
        Log.w("Create Channel", String.format("Set lockscreen %d", channel.getLockscreenVisibility()));

        notificationManager.createNotificationChannel(channel);
    }

    protected void printNotificationChannel(NotificationManager notificationManager)
    {
        NotificationChannel channel = notificationManager.getNotificationChannel(k_ChannelId);
        Log.w("Print Channel", String.format("Get lockscreen %d", channel.getLockscreenVisibility()));
    }
```
I got the below results
```
2020-04-13 16:52:38.320 17408-17408/com.example.testvibration W/Create Channel: Default lockscreen: -1000
2020-04-13 16:52:38.321 17408-17408/com.example.testvibration W/Create Channel: Set lockscreen 0
2020-04-13 16:52:49.375 17408-17408/com.example.testvibration W/Print Channel: Get lockscreen -1000
```
Then I went to System Settings to change the lockscreen visibility to `Hide senstitive ....`(private), I got
```
2020-04-13 16:57:36.180 17408-17408/com.example.testvibration W/Print Channel: Get lockscreen 0
```
At last I switched back to `Show all notification contents`, I got,
```
2020-04-13 16:59:36.190 15045-15045/com.example.testvibration W/Print Channel: Get lockscreen -1000
``` 

So my conclusions are 
- you can't really set the lockscreen visibility from code...
- -1000 stands for public

And my question is do we really provide LockScreenVisibility setter for NotificationChannel...

**Comment by Maicolo** [comment]
Interesting, thanks for the heads up Vincent!

If I use AndroidNotificationCenter.RegisterNotificationChannel passing a channel where 'LockScreenVisibility' has been seto to 'LockScreenVisibility.Secret', the channel seems to work as intended. I can't see any notification fired on that channel in the lock screen.

**Comment by vincent-unity** [comment]
So weird... We have tests on NotificationChannel which all the returned value of NotificationChannel.LockscreenVisibility are Public, no matter which value you set on creation. I'm personally testing on Google Pixel 2 with Android 8.

So could you please check what NotificationChannel.LockscreenVisiblity returns for a registered channel? Please use AndroidNotificationCenter.GetNotificationChannel() to retreive the registered channel. I'm expecting it returns Private (which should be Public as I'm going to fix).

**Comment by Maicolo** [comment]
Sorry for the late answer Vincent,
so I have performed the test you asked me. To be clear I am using the version 1.0.3 of the mobile notification package and I am testing on a very old KitKat Android phone.

After registering three different channels with the three different visibility values this is the output:

>I/Unity: The visibility for the channel 'game_normal_channel' is: Public
>I/Unity: The visibility for the channel 'game_private_channel' is: Private
>I/Unity: The visibility for the channel 'game_silent_channel' is: Secret

I confirm that the 'Public' and the 'Secret' channel work just fine while the 'Private' one bahave like the 'Public' one.

**Comment by vincent-unity** [comment]
Thanks! So it's a different story on device below Android 8, as there is no notification channel support.

**Comment by vincent-unity** [comment]
Fixed in 1.2.0-preview.
